core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -93,17 +92,26 @@ public Sep31PostTransactionResponse postTransaction(
 
     validateKyc(request.getSenderId(), request.getReceiverId());
     validateAndApplyFeeAndQuote(txn, request, jwtToken);
+    generateTransactionMemo(txn);
 
     sep31TransactionStore.save(txn);
 
     return Sep31PostTransactionResponse.builder()
         .id(txn.getId())
-        .stellarAccountId(txn.getStellarAccountId())
+        .stellarAccountId(asset.getDistributionAccount())
         .stellarMemo(txn.getStellarMemo())
-        .stellarMemo(txn.getStellarMemoType())
+        .stellarMemoType(txn.getStellarMemoType())
         .build();
   }
 
+  private void generateTransactionMemo(Sep31Transaction txn) throws SepException {
+    String memo = StringUtils.truncate(txn.getId(), 32);
+    memo = StringUtils.leftPad(memo, 32, '0');
+    memo = new String(Base64.getEncoder().encode(memo.getBytes()));
+    txn.setStellarMemo(memo);
+    txn.setStellarMemoType(memoType(MEMO_HASH));
+  }
+
```

Comment:
I think we should make the memo values less predictable. Nowadays I believe our IDs are incrementally generated, so we should probably update our id generator strategy to UUID by changing [this](https://github.com/stellar/java-stellar-anchor-sdk/blob/9df535a5712b9645a0298a917ee6d8de40641b6d/data-spring-jdbc/src/main/java/org/stellar/anchor/server/data/JdbcSep31Transaction.java#L22) to:

```
@Id
@GeneratedValue
UUID id;
```

If we don't, someone could mess with our watchers by sending 1 stroop payments, for instance.